rotkehlchen/globaldb/handler.py

```diff
@@ -250,24 +250,25 @@ def retrieve_assets(filter_query: 'AssetsFilterQuery') -> Tuple[List[Dict[str, A
         underlying_tokens: Dict[str, List[Dict[str, str]]] = defaultdict(list)
         prepared_filter_query, bindings = filter_query.prepare()
         parent_query = """
-        SELECT A.identifier AS identifier, A.type, B.address, B.decimals, A.name, C.symbol, C.started, C.forked, C.swapped_for, C.coingecko, C.cryptocompare, B.protocol, B.chain, B.token_kind, D.notes, D.type AS custom_asset_type
-        FROM assets as A
+        SELECT A.identifier AS identifier, A.type, B.address, B.decimals, A.name, C.symbol,
+        C.started, C.forked, C.swapped_for, C.coingecko, C.cryptocompare, B.protocol, B.chain,
+        B.token_kind, D.notes, D.type AS custom_asset_type FROM
+        assets as A
         LEFT JOIN common_asset_details AS C ON C.identifier = A.identifier
         LEFT JOIN evm_tokens as B ON B.identifier = A.identifier
         LEFT JOIN custom_assets as D ON D.identifier = A.identifier
-        """  # noqa: E501
+        """
         query = f'SELECT * FROM ({parent_query}) {prepared_filter_query}'
-        underlying_tokens_query = """
-        SELECT assets.identifier AS identifier, address, token_kind, parent_token_entry, weight, name, symbol, type, chain
-        FROM underlying_tokens_list
-        INNER JOIN assets ON assets.identifier = parent_token_entry
-        INNER JOIN evm_tokens ON assets.identifier=evm_tokens.identifier
-        INNER JOIN common_asset_details ON common_asset_details.identifier = assets.identifier
-        """  # noqa: E501
+        underlying_tokens_query = (
+            f'SELECT parent_token_entry, address, token_kind, weight FROM underlying_tokens_list '
+            f'LEFT JOIN evm_tokens ON underlying_tokens_list.identifier=evm_tokens.identifier '
+            f'WHERE parent_token_entry IN (SELECT identifier FROM ({query}))'
+        )
+
         with GlobalDBHandler().conn.read_ctx() as cursor:
             # get all underlying tokens
-            for entry in cursor.execute(f'SELECT * FROM ({underlying_tokens_query}) {prepared_filter_query}', bindings):  # noqa: E501
-                underlying_tokens[entry[0]].append(UnderlyingToken.deserialize_from_db((entry[1], entry[2], entry[4])).serialize())  # noqa: E501
+            for entry in cursor.execute(f'SELECT * FROM ({underlying_tokens_query})', bindings):  # noqa: E501
```

Comment:
This part of the code is really complicated and to be hnest I don't get any of these queries very well. Can't they be written in a better way?

For example why this complexity here?

```python
f'SELECT * FROM ({underlying_tokens_query})', bindings
```
Can't you just use the actual query? Why do you do SELECT * from the other query again?

Reply:
Indeed was my fault, is not needed the `SELECT from` and is there from the previous query. Regarding the queries I'll explain what each one do